x-pack/plugins/lens/public/visualizations/datatable/components/dimension_editor.tsx

```diff
@@ -45,43 +48,54 @@ function updateColumnWith(
 export function TableDimensionEditor(
   props: VisualizationDimensionEditorProps<DatatableVisualizationState> & {
     paletteService: PaletteRegistry;
+    isDarkMode: boolean;
   }
 ) {
-  const { state, setState, frame, accessor, isInlineEditing } = props;
-  const column = state.columns.find(({ columnId }) => accessor === columnId);
+  const { frame, accessor, isInlineEditing, isDarkMode } = props;
+  const column = props.state.columns.find(({ columnId }) => accessor === columnId);
+  const { inputValue: localState, handleInputChange: setLocalState } =
+    useDebouncedValue<DatatableVisualizationState>({
+      value: props.state,
+      onChange: props.setState,
+    });
+
+  const updateColumnState = useCallback(
+    (columnId: string, newColumn: Partial<ColumnType>) => {
+      setLocalState({
+        ...localState,
+        columns: updateColumn(localState, columnId, newColumn),
+      });
+    },
+    [setLocalState, localState]
+  );
 
   if (!column) return null;
   if (column.isTransposed) return null;
 
-  const currentData = frame.activeData?.[state.layerId];
-
-  // either read config state or use same logic as chart itself
-  const isNumeric = isNumericFieldForDatatable(currentData, accessor);
+  const currentData = frame.activeData?.[localState.layerId];
+  const dataType = getFieldTypeFromDatatable(currentData, accessor);
+  const isNumeric = dataType === 'number';
   const currentAlignment = column?.alignment || (isNumeric ? 'right' : 'left');
   const currentColorMode = column?.colorMode || 'none';
   const hasDynamicColoring = currentColorMode !== 'none';
+  const showDynamicColoringFeature = dataType !== 'date';
+  const visibleColumnsCount = localState.columns.filter((c) => !c.hidden).length;
 
-  const datasource = frame.datasourceLayers[state.layerId];
-  const showDynamicColoringFeature = Boolean(
-    isNumeric && !datasource?.getOperationForColumnId(accessor)?.isBucketed
-  );
-
-  const visibleColumnsCount = state.columns.filter((c) => !c.hidden).length;
-
-  const hasTransposedColumn = state.columns.some(({ isTransposed }) => isTransposed);
+  const hasTransposedColumn = localState.columns.some(({ isTransposed }) => isTransposed);
   const columnsToCheck = hasTransposedColumn
     ? currentData?.columns.filter(({ id }) => getOriginalId(id) === accessor).map(({ id }) => id) ||
       []
     : [accessor];
   const minMaxByColumnId = findMinMaxByColumnId(columnsToCheck, currentData, getOriginalId);
   const currentMinMax = minMaxByColumnId[accessor];
 
-  const activePalette = column?.palette || {
+  const activePalette = column?.palette ?? {
     type: 'palette',
     name: defaultPaletteParams.name,
```

Comment:
The default palette here defined is `positive` and its used for both color by value and the new color mapping.
I have few questions about that, as it looks weird to me to use a sequential for a categorical use case.

![table_default_palette_mapping_old](https://github.com/user-attachments/assets/e6508a5b-827e-49e0-93ee-6e0713d75bea)

it makes more sense to me to use, in the case of the color mapping, the `default` palette (as it is for other chart types):

![table_default_palette_mapping](https://github.com/user-attachments/assets/80140920-0f78-4f5a-8eea-e3fcf515a40a)

Maybe a lighter hue version of the palette would work even better in the cell use case (but not a big deal here). cc @MichaelMarcialis 

The suggested change is minimal here:
```suggestion
    name: isNumerical ? defaultPaletteParams.name : 'default',
```

Reply:
Yeah I agree, I just noticed that this week.